### PR TITLE
add support for modbus function 17

### PR DESCRIPTION
**Pull request by nusaqib**
Added support for function 17. I tested it with Modicon PLC series 984 and it is working fine with no errors. Please review!

### CODE REVIEW

**Review by MarkRivers — CHANGES_REQUESTED** [review]
This looks great, thanks!

I have 2 requests:
- The Modbus documentation calls F17 "Report Slave ID".  I think it would be good to globally change READ_SLAVE_ID to REPORT_SLAVE_ID for consistency.
- You have inserted blank lines where nearby code does not have those.  Can you remove those for stylistic consistence?

Since F17 can return a variable number of bytes how does the user specify how many bytes to expect?  What record is used for the response?

Does it make sense to include an example startup script that shows how to use F17?

**Comment by nusaqib** [comment]
-  Suggested changes are incorporated
-  Added sample cmd and substitutions files
- modbusLength defines device-specific number of bytes to be received
- I am using either mbbiDirect or longin. mbbiDirect is more suitable as different bits have different meanings.

File: modbusApp/src/modbus.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #define MODBUS_WRITE_SINGLE_REGISTER         0x06
 #define MODBUS_WRITE_MULTIPLE_COILS          0x0F
 #define MODBUS_WRITE_MULTIPLE_REGISTERS      0x10
+#define MODBUS_REPORT_SLAVE_ID                 0x11
```
Comment (nusaqib) [inline modbusApp/src/modbus.h:22]: @MarkRivers sorry for this misalignment, should I do another commit for aligning this?

**Comment by prjemian** [comment]
> * The Modbus documentation calls F17 "Report Slave ID".  I think it would be good to globally change READ_SLAVE_ID to REPORT_SLAVE_ID for consistency.

Thanks for this info.  Was missing this in the PR's description.

File: modbusApp/src/modbus.h (hunk 1 above)
Comment (MarkRivers) [reply modbusApp/src/modbus.h:22]: I will fix it after merging.

**Comment by MarkRivers** [comment]
I made some minor changes, and added updates to the documentation to include function 17.
